File: Scripts/server/regular-expressions/c++/regular-expressions.cpp
#include <iostream>
#include <string>
#include <regex>

int main ()
{
  volatile bool m;

  for ( int i = 0; i < 10000000; ++i ) { 
  std::string base="this is a test string";
  std::string str=base; 

    m = std::regex_match (base, std::regex("(.*)(string)"));
    str.replace(str.begin()+15,str.begin()+23,"istyfied"); 
  }

  return 0;
}

File: Scripts/server/regular-expressions/c-sharp/regular-expressions.cs
using System;
using System.Text.RegularExpressions;

class Program {
    
    private static volatile int r;

    public static int executeTask(int i) {
	
	
        string str = "this is a matching string";

        new Regex("string$").IsMatch(str);
        new Regex("string").Replace(str, "istyfied");

	return 1 + i;
    }
 
    static void Main(string[] args) {
    
    for (int i = 0; i < 10000000; ++i) {
    		r = executeTask(i + r);
	}
    }
}
